### ISSUE TITLE
[FEATURE] PHP version statistics

### ISSUE THREAD

**Issue by vimishor** [composer/packagist#482]
Because composer includes PHP version in User-Agent, maybe this data could be made available on the stats page ?

I don't know if this data is being collected at this point, but I think that it will be a nice addition to the stats page.

I think that seeing some usage stats about each PHP version could help the community and in special package maintainers, which could make a better decision when picking the minimum required PHP version for their package.

**Comment by vimishor** [comment]
/ping @Seldaek Is this data being collected at this point ? Do you think this is feasible ?

**Comment by Seldaek** [comment]
It's not exactly collected right now except in the access logs of the web server. I try to run stats on that at least once yearly to show the evolution.. see http://seld.be/notes/my-view-of-php-version-adoption

Collecting it more than that would just mean parsing the access logs and storing the data somewhere. It's not impossible but I am not sure how relevant this is anyway given that the results are fairly skewed towards dev envs.

**Comment by vimishor** [comment]
@Seldaek I know about your blog post. Actually reading that post I got this idea.

I wasn't thinking about storing the data in DB or something similar, but more if your think that is feasible to parse that data from the logs once each year for example and publish the results on the stats page from packagist.

So everything published on packagist is just static data, updated yearly.

**Comment by Seldaek** [comment]
I see, yeah I guess we could do it even monthly if it's scripted.. Problem is doing it :)

**Comment by alexbowers** [comment]
@Seldaek Could you post a gist with an excerpt of the logs? Maybe somebody will push some code which parses it.

Thanks

**Comment by Seldaek** [comment]
It's standard nginx logs like:

```
123.123.123.123 - - [20/Jun/2015:16:54:39 +0000] "GET /p/provider-2015-04$5603e22c8afb907f619acd8c880eeb6035696ab420dd1e1987befdc8a1c86c1d.json HTTP/1.1" 200 2959257 "-" "Composer/9fb2d4f2d642a0749decb41bc2fe4be2bf8bef7a (Linux; 3.13.0-24-generic; PHP 5.5.9)"
```

It's not really hard to parse though, it's more about figuring out how to store and represent it in a useful way.

**Comment by alexbowers** [comment]
Can you provide me with a relatively large zipped version of the logs? Maybe going back a month or so, and i'll give it a go :)

**Comment by Seldaek** [comment]
Sure I'll email you that

**Comment by alexbowers** [comment]
The point im up to at the moment is the regular expression works;

```
^(?:[^;\n]*;){2}\s+(?P<php_version>[^\)]+)
```

When passed the value of some of the log, it matches the correct versions perfectly. However, I am now struggling with putting it into part of a sed command.

The plan was to get sed to run the regular expression, and replace the lines with the match (So the file will be filled with PHP 5.4.0, one per line for example).

Does anybody have any advice on this?

https://regex101.com/r/eO2wG4/2

**Comment by Seldaek** [comment]
@alexbowers did you check the stuff I sent you? It included all the sed & so on to output a full list of version + count of requests

**Comment by Seldaek** [comment]
This is finally available :) See https://blog.packagist.com/sunsetting-the-php-version-stats-blog-series/ for details and sample links.
